wokspace-loader/src/index.ts

```diff
@@ -14,64 +14,66 @@ const css = require('./style.css');
 import { WebsocketClient } from './json-rpc/websocket-client';
 import { CheJsonRpcMasterApi } from './json-rpc/che-json-rpc-master-api';
 import { getWorkspace, startWorkspace } from './workspace';
+import { Loader } from './loader/loader'
 
-const DEFAULT_WEBSOCKET_CONTEXT = '/api/websocket';
+const WEBSOCKET_CONTEXT = '/api/websocket';
 
-let wsClient = new WebsocketClient();
-let wsUrl = formJsonRpcApiLocation(document.location) + DEFAULT_WEBSOCKET_CONTEXT;
-let masterApi = new CheJsonRpcMasterApi(wsClient);
-let workspaceKey = getWorkspaceKey();
-masterApi.connect(wsUrl).then(() => {
-    getWorkspace(getWorkspaceKey()).then(handleWorkspace).catch(err => {
-        console.error(err);
-    });
+let websocketURL = websocketBaseURL() + WEBSOCKET_CONTEXT;
+let master = new CheJsonRpcMasterApi(new WebsocketClient());
+let loader = new Loader();
+
+master.connect(websocketURL).then(() => {
+    getWorkspace(getWorkspaceKey())
+        .then(handleWorkspace)
+        .catch(err => {console.error(err);});
 });
 
-function handleWorkspace(ws: che.IWorkspace) {
-    let machineStatusHandler = (message: any) => {
-        console.log('machine status', message);
-    };
-    let machineOutputHandler = (message: any) => {
-        document.getElementById("output").appendChild(document.createTextNode(message.text));
-    };
-    let workspaceStatusHandler = (message: any) => {
-        console.log('workspace status', message);
-        let workspaceStatus = message.status;
-
-        if (workspaceStatus === 'RUNNING') {
-            openIdeAfterRunning(ws);
-        }
+function handleWorkspace(workspace: che.IWorkspace) {
```

Comment:
it's not related to this PR but it would be great to have such functions moved classes

Reply:
I don't understand pretty sure, where we should move those functions.

Reply:
it looks like a WsMaster connection so it could be a WsMaster class or WebScoketWsMaster